editor/themes/editor_theme_manager.cpp

```diff
@@ -0,0 +1,2232 @@
+	config.color_picker_button_height = 28 * EDSCALE;
+
+	config.default_contrast = 0.3; // Make sure to keep this in sync with the editor settings definition.
+
+	// Handle main theme preset.
+	{
+		if (config.preset != "Custom") {
+			Color preset_accent_color;
+			Color preset_base_color;
+			float preset_contrast = 0;
+			bool preset_draw_extra_borders = false;
+
+			// Please use alphabetical order if you're adding a new theme here.
+			if (config.preset == "Breeze Dark") {
+				preset_accent_color = Color(0.26, 0.76, 1.00);
+				preset_base_color = Color(0.24, 0.26, 0.28);
+				preset_contrast = config.default_contrast;
+			} else if (config.preset == "Godot 2") {
+				preset_accent_color = Color(0.53, 0.67, 0.89);
+				preset_base_color = Color(0.24, 0.23, 0.27);
+				preset_contrast = config.default_contrast;
+			} else if (config.preset == "Gray") {
+				preset_accent_color = Color(0.44, 0.73, 0.98);
+				preset_base_color = Color(0.24, 0.24, 0.24);
+				preset_contrast = config.default_contrast;
+			} else if (config.preset == "Light") {
+				preset_accent_color = Color(0.18, 0.50, 1.00);
+				preset_base_color = Color(0.9, 0.9, 0.9);
+				// A negative contrast rate looks better for light themes, since it better follows the natural order of UI "elevation".
+				preset_contrast = -0.06;
+			} else if (config.preset == "Solarized (Dark)") {
+				preset_accent_color = Color(0.15, 0.55, 0.82);
+				preset_base_color = Color(0.04, 0.23, 0.27);
+				preset_contrast = config.default_contrast;
+			} else if (config.preset == "Solarized (Light)") {
+				preset_accent_color = Color(0.15, 0.55, 0.82);
+				preset_base_color = Color(0.89, 0.86, 0.79);
+				// A negative contrast rate looks better for light themes, since it better follows the natural order of UI "elevation".
+				preset_contrast = -0.06;
+			} else if (config.preset == "Black (OLED)") {
+				preset_accent_color = Color(0.45, 0.75, 1.0);
+				preset_base_color = Color(0, 0, 0);
+				// The contrast rate value is irrelevant on a fully black theme.
+				preset_contrast = 0.0;
+				preset_draw_extra_borders = true;
+			} else { // Default
+				preset_accent_color = Color(0.44, 0.73, 0.98);
+				preset_base_color = Color(0.21, 0.24, 0.29);
+				preset_contrast = config.default_contrast;
+			}
+
+			config.accent_color = preset_accent_color;
+			config.base_color = preset_base_color;
+			config.contrast = preset_contrast;
+			config.draw_extra_borders = preset_draw_extra_borders;
+
+			EditorSettings::get_singleton()->set_initial_value("interface/theme/accent_color", config.accent_color);
+			EditorSettings::get_singleton()->set_initial_value("interface/theme/base_color", config.base_color);
+			EditorSettings::get_singleton()->set_initial_value("interface/theme/contrast", config.contrast);
+			EditorSettings::get_singleton()->set_initial_value("interface/theme/draw_extra_borders", config.draw_extra_borders);
```

Comment:
I would put this into a function.

Reply:
What would be the benefit compared to it just being scoped?

Reply:
It's just that I find that it's a long if statement. But yeah, you're right.